Sources/OpenAPIRuntime/Conversion/Configuration.swift

```diff
@@ -25,14 +25,45 @@ public protocol DateTranscoder: Sendable {
 }
 
 /// A transcoder for dates encoded as an ISO-8601 string (in RFC 3339 format).
-public struct ISO8601DateTranscoder: DateTranscoder {
+public struct ISO8601DateTranscoder: DateTranscoder, @unchecked Sendable {
+
+    /// The lock protecting the formatter.
+    private let lock: NSLock
+
+    /// The underlying date formatter.
+    private let locked_formatter: ISO8601DateFormatter
+
+    /// Creates a new default transcoder.
+    public init() { self.init(optionsOverride: nil) }
+
+    /// Creates a new transcoder with the provided options.
+    /// - Parameter optionsOverride: Options to override the default ones. If you provide
+    ///   nil here, the default options are used.
+    public init(optionsOverride: ISO8601DateFormatter.Options?) {
+        let formatter = ISO8601DateFormatter()
+        if let optionsOverride { formatter.formatOptions = optionsOverride }
+        self.init(formatter: formatter)
+    }
+
+    /// Creates a new transcoder with the provided formatter.
+    /// - Parameter formatter: The formatter to use.
+    public init(formatter: ISO8601DateFormatter) {
+        lock = NSLock()
+        locked_formatter = formatter
+    }
 
     /// Creates and returns an ISO 8601 formatted string representation of the specified date.
-    public func encode(_ date: Date) throws -> String { ISO8601DateFormatter().string(from: date) }
+    public func encode(_ date: Date) throws -> String {
+        lock.lock()
+        defer { lock.unlock() }
```

Comment:
If we do decide to keep the formatter and still need locking can we pull in the `LockedValueBox` so we don't need to use unchecked sendable?

Reply:
Sure, we use NSLock in other places as well so we should probably migrate them all to LockedValueBox if we introduce it to the runtime library.

Reply:
Tracked by https://github.com/apple/swift-openapi-generator/issues/505.